bin/terminal.js

```diff
@@ -36,21 +36,53 @@ function listPorts() {
   });
 };
 
-function createPort() {
-  if (!args.port) {
-    args.outputHelp();
-    args.missingArgument('port');
-    process.exit(-1);
+function setupPort() {
+  if(args.port) {
+    createPort(args.port);
   }
 
+  SerialPort.list((err, ports) => {
+    if (err) {
+      console.error('Error listing ports, and missing port argument.', err);
+      args.outputHelp();
+      args.missingArgument('port');
+      process.exit(-1);
+    } else {
+      if(ports.length > 0) {
+        var portSelection = new List({
+          name: 'serial-port-selection',
+          message: 'Select a serial port to open',
+          choices: ports.map((port, i) => `[${i + 1}]\t${port.comName}\t${port.pnpId || ''}\t${port.manufacturer || ''}`)
+        });
+  
+        portSelection.run()
+          .then(answer => {
+            const choice = answer.split('\t')[1];
```

Comment:
This is probably the hackiest part of the whole PR. Relying on the choices to be tab separated in order to extract the comName.

I would prefer it if the prompt-list library had a `name` and `value` fields so that what you display could be different to what gets returned as an 'answer' but, alas, it does not look like that is the case.